Description: work around latex tabu bug
Last-Update: 2019-02-01
Forwarded: https://bugs.debian.org/920621
Author: Paolo Greppi <paolo.greppi@libpf.com>

Updated: 2024-09-13 for MacPorts, target doxygen 1.12.0
Updated: 2025-05-28 for MacPorts, target doxygen 1.14.0

Index: doxygen/src/latexdocvisitor.cpp
===================================================================
--- src/latexdocvisitor.cpp.orig	2025-05-24 04:50:50
+++ src/latexdocvisitor.cpp	2025-05-28 08:08:16
@@ -1257,7 +1257,7 @@
   }
   else
   {
-    m_t << "\\tabulinesep=1mm\n\\begin{longtabu}spread 0pt [c]{*{" << cols << "}{|X[-1]}|}\n";
+    m_t << "\\tabulinesep=1mm\n\\begin{longtabu}spread 0pt [c]{*{" << cols << "}{|p{1cm}}|}\n";
   }
   //return isNested ? "TabularNC" : "TabularC";
 }

Index: doxygen/templates/latex/doxygen.sty
===================================================================
--- templates/latex/doxygen.sty.orig	2025-05-15 13:16:04
+++ templates/latex/doxygen.sty	2025-05-28 08:08:53
@@ -351,10 +351,10 @@
     \tabulinesep=1mm%
     \par%
     \ifthenelse{\equal{#1}{}}%
-      {\begin{longtabu*}spread 0pt [l]{|X[-1,l]|X[-1,l]|}}% name + description
+      {\begin{longtabu*}spread 0pt [l]{|c|p{1cm}|}}% name + description
     {\ifthenelse{\equal{#1}{1}}%
-      {\begin{longtabu*}spread 0pt [l]{|X[-1,l]|X[-1,l]|X[-1,l]|}}% in/out + name + desc
-      {\begin{longtabu*}spread 0pt [l]{|X[-1,l]|X[-1,l]|X[-1,l]|X[-1,l]|}}% in/out + type + name + desc
+      {\begin{longtabu*}spread 0pt [l]{|c|c|p{1cm}|}}% in/out + name + desc
+      {\begin{longtabu*}spread 0pt [l]{|c|c|c|p{1cm}|}}% in/out + type + name + desc
     }
     \multicolumn{2}{l}{\hspace{-6pt}\bfseries\fontseries{bc}\selectfont\color{darkgray} #2}\\[1ex]%
     \hline%
@@ -371,7 +371,7 @@
 \newenvironment{DoxyFields}[1]{%
     \tabulinesep=1mm%
     \par%
-    \begin{longtabu*}spread 0pt [l]{|X[-1,r]|X[-1,l]|X[-1,l]|}%
+    \begin{longtabu*}spread 0pt [l]{|c|c|c|}%
     \multicolumn{3}{l}{\hspace{-6pt}\bfseries\fontseries{bc}\selectfont\color{darkgray} #1}\\[1ex]%
     \hline%
     \endfirsthead%
@@ -388,8 +388,8 @@
     \tabulinesep=1mm%
     \par%
     \ifthenelse{\equal{#1}{2}}%
-      {\begin{longtabu*}spread 0pt [l]{|X[-1,r]|X[-1,l]|}}%
-      {\begin{longtabu*}spread 0pt [l]{|X[-1,l]|X[-1,r]|X[-1,l]|}}% with init value
+      {\begin{longtabu*}spread 0pt [l]{|c|p{1cm}|}%
+      {\begin{longtabu*}spread 0pt [l]{|c|c|p{1cm}|}}% with init value
     \multicolumn{2}{l}{\hspace{-6pt}\bfseries\fontseries{bc}\selectfont\color{darkgray} #2}\\[1ex]%
     \hline%
     \endfirsthead%
@@ -411,7 +411,7 @@
 \newenvironment{DoxyRetVals}[1]{%
     \tabulinesep=1mm%
     \par%
-    \begin{longtabu*}spread 0pt [l]{|X[-1,r]|X[-1,l]|}%
+    \begin{longtabu*}spread 0pt [l]{|c|c|}%
     \multicolumn{2}{l}{\hspace{-6pt}\bfseries\fontseries{bc}\selectfont\color{darkgray} #1}\\[1ex]%
     \hline%
     \endfirsthead%
@@ -427,7 +427,7 @@
 \newenvironment{DoxyExceptions}[1]{%
     \tabulinesep=1mm%
     \par%
-    \begin{longtabu*}spread 0pt [l]{|X[-1,r]|X[-1,l]|}%
+    \begin{longtabu*}spread 0pt [l]{|c|c|}%
     \multicolumn{2}{l}{\hspace{-6pt}\bfseries\fontseries{bc}\selectfont\color{darkgray} #1}\\[1ex]%
     \hline%
     \endfirsthead%
@@ -443,7 +443,7 @@
 \newenvironment{DoxyTemplParams}[1]{%
     \tabulinesep=1mm%
     \par%
-    \begin{longtabu*}spread 0pt [l]{|X[-1,r]|X[-1,l]|}%
+    \begin{longtabu*}spread 0pt [l]{|c|c|}%
     \multicolumn{2}{l}{\hspace{-6pt}\bfseries\fontseries{bc}\selectfont\color{darkgray} #1}\\[1ex]%
     \hline%
     \endfirsthead%
@@ -519,11 +519,11 @@
 \newcommand{\PBS}[1]{\let\temp=\\#1\let\\=\temp}%
 \newenvironment{TabularC}[1]%
 {\tabulinesep=1mm
-\begin{longtabu*}spread 0pt [c]{*#1{|X[-1]}|}}%
+\begin{longtabu*}spread 0pt [c]{*#1{|c}|}}%
 {\end{longtabu*}\par}%
 
 \newenvironment{TabularNC}[1]%
-{\begin{tabu}spread 0pt [l]{*#1{|X[-1]}|}}%
+{\begin{tabu}spread 0pt [l]{*#1{|c}|}}%
 {\end{tabu}\par}%
 
 % Used for member group headers
